src/main/java/ch/wisv/chue/hue/PhilipsHueFacade.java

```diff
@@ -236,4 +242,54 @@ public void updateLightState(HueLamp lamp, HueLightState lightState) throws Brid
     public boolean isBridgeAvailable() {
         return bridge != null;
     }
+
+    /**
+     * Check for Hue software update, and install it if available.
+     * <p>
+     * http://www.developers.meethue.com/documentation/software-update
+     */
+    @Scheduled(cron = "54 54 2 * * *")
+    public void updateBridgeFirmware() {
+        RestTemplate rt = new RestTemplate();
+        String bridgeConfigUrl = buildBridgeHttpAddress() + "config";
+        HueSWUpdate swUpdate = rt.getForEntity(bridgeConfigUrl, HueConfig.class).getBody().getSWUpdate();
+
+        if (swUpdate.isNotify()) {
+            log.info("Bridge update completed");
+            rt.put(bridgeConfigUrl, "{\"swupdate\": {\"notify\":false}}");
+        } else if (swUpdate.getUpdatestate() == 0) {
+            log.info("Initiating bridge update check");
+            rt.put(bridgeConfigUrl, "{\"swupdate\": {\"checkforupdate\":true}}");
+        }
+        if (swUpdate.getUpdatestate() == 2) {
+            log.info("Installing bridge update: {}", swUpdate.getText());
+            rt.put(bridgeConfigUrl, "{\"swupdate\": {\"updatestate\":3}}");
+        }
```

Comment:
The `PHHueHttpConnection` class isn't documented so that's why I didn't want to use it.